editor/src/marionette/preview.ts

```diff
@@ -0,0 +1,357 @@
+import { BlendStateBuffer, LegacyBlendStateBuffer } from '../../../cocos/3d/skeletal-animation/skeletal-animation-blending';
+import { Node } from '../../../cocos/core';
+import { Motion, MotionEval } from '../../../cocos/core/animation/marionette/motion';
+import { createEval } from '../../../cocos/core/animation/marionette/create-eval';
+import { BindContext } from '../../../cocos/core/animation/marionette/parametric';
+import { VarInstance, Value, VariableType } from '../../../cocos/core/animation/marionette/variable';
+import { assertIsNonNullable } from '../../../cocos/core/data/utils/asserts';
+import {
+    AnimationBlendEval,
+} from '../../../cocos/core/animation/marionette/animation-blend';
+import type { RuntimeID } from '../../../cocos/core/animation/marionette/graph-debug';
+
+class AnimationGraphPartialPreviewer {
+    constructor(root: Node) {
+        this._root = root;
+    }
+
+    public destroy() {
+    }
+
+    public evaluate() {
+        this._blendBuffer.apply();
+    }
+
+    public addVariable(id: string, type: VariableType, value: Value) {
+        this._varInstances[id] = new VarInstance(type, value);
+    }
+
+    public removeVariable(id: string) {
+        delete this._varInstances[id];
+    }
+
+    public updateVariable(id: string, value: Value) {
+        this._varInstances[id].value = value;
```

Comment:
Need to check if variable exists?

Reply:
Solved in 7a7bb82049ffe5b5eab6695a2f1a7ad6bda1879d